tests/ga/test_exthandlers.py

```diff
@@ -249,6 +247,22 @@ def test_extension_sequence_number(self):
                                               disk_sequence_number=3,
                                               expected_sequence_number=-1)
 
+    def test_it_should_report_error_if_plugin_settings_version_mismatch(self):
+        with mock_wire_protocol(mockwiredata.DATA_FILE_PLUGIN_SETTINGS_MISMATCH) as protocol:
+            with patch("azurelinuxagent.common.protocol.goal_state.add_event") as mock_add_event:
+                # Forcing update of GoalState to allow the ExtConfig to report an event
+                protocol.mock_wire_data.set_incarnation(2)
```

Comment:
The `protocol.detect()` function inside `mock_wire_protocol()` does a forced `update_goal_state()` but for the purpose of this test, I did not want to rely on that as that behavior so doing an incarnation change and updating the GS to test out the behavior instead.